polaris-react/src/components/Select/Select.tsx

```diff
@@ -16,7 +16,7 @@ interface StrictOption {
   /** Machine value of the option; this is the value passed to `onChange` */
   value: string;
   /** Human-readable text for the option */
-  label: string;
+  label: string | React.ReactNode;
```

Comment:
```suggestion
  label: React.ReactNode;
```
[ReactNode](https://reactnative.dev/docs/react-node) includes `string` as type so we don't need to explicitly include string.